src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/rca/persistence/FileRotate.java

```diff
@@ -0,0 +1,129 @@
+ *  A copy of the License is located at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  or in the "license" file accompanying this file. This file is distributed
+ *  on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ *  express or implied. See the License for the specific language governing
+ *  permissions and limitations under the License.
+ */
+
+package com.amazon.opendistro.elasticsearch.performanceanalyzer.rca.persistence;
+
+import java.io.IOException;
+import java.nio.file.FileAlreadyExistsException;
+import java.nio.file.Files;
+import java.nio.file.NoSuchFileException;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.text.DateFormat;
+import java.util.concurrent.TimeUnit;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+public class FileRotate {
+  private static final Logger LOG = LogManager.getLogger(FileRotate.class);
+
+  private final Path FILE_TO_ROTATE;
+  private final String FILENAME;
+  private final TimeUnit ROTATION_TIME_UNIT;
+  private final long ROTATION_PERIOD;
+  private final DateFormat ROTATED_FILE_FORMAT;
+  protected long lastRotatedMillis;
+  private static final String FILE_PART_SEPARATOR = ".";
+
+  FileRotate(
+      Path file_to_rotate,
+      TimeUnit rotation_time_unit,
+      long rotation_period,
+      DateFormat rotated_file_format) {
+    FILE_TO_ROTATE = file_to_rotate;
+    FILENAME = file_to_rotate.getFileName().toString();
+    ROTATION_TIME_UNIT = rotation_time_unit;
+    ROTATION_PERIOD = rotation_period;
+    ROTATED_FILE_FORMAT = rotated_file_format;
+    lastRotatedMillis = System.currentTimeMillis();
+  }
+
+  /**
+   * Try to rotate the file.
+   *
+   * <p>The file is rotated only if the current time is past the ROTATION_PERIOD.
+   *
+   * @return null if the file was not rotated because it is not old enough, or the name of the file
+   *     after rotation.
+   */
+  Path tryRotate(long currentTimeMillis) throws IOException {
+    if (shouldRotate(currentTimeMillis)) {
+      return rotate(currentTimeMillis);
+    }
+    return null;
+  }
+
+  /**
+   * This does not check for the validity of the condition for rotation. Just rotates the file.
+   *
+   * <p>This can be called when say, the DB file is corrupted and we just want to rotate and create
+   * a new file instead of throwing away data in this iteration.
+   *
+   * @return Path to the file after it is rotated or null if it ran into a problem trying to do so.
+   */
+  Path forceRotate(long currentTimeMillis) throws IOException {
+    return rotate(currentTimeMillis);
+  }
+
+  /**
+   * This checks for the condition for file rotation.
+   *
+   * <p>This usually checks if the time since last rotation is past the {@code ROTATION_PERIOD}.
+   *
+   * @return
+   */
+  protected boolean shouldRotate(long currentTimeMillis) {
+    long timeSinceLastRotation = currentTimeMillis - lastRotatedMillis;
```

Comment:
Just curious, we are rotating file on the basis of time passed or should we rotate if the current time hour > file hour

for example: at 4:00PM, we should rotate file which is even created at 3.30
It should happen as we are rotating file at 4.30.

Had a discussion regarding this with Karthik

Reply:
We are rotating based on 1 hour since the file creation.

Reply:
I will probably add a subsequent patch to have this change. Don't want to make this PR bigger.